feat: Improve guide connecting machines and creating dashboards (#678)
united-manufacturing-hub/united-manufacturing-hub: docs/content/en/docs/Getting started/connecting-machines-creating-dashboards/index.md

  This section explains how the United Manufacturing Hub is used practically 
---

## 0. Video: From sensor to dashboard

TODO
#455

## 1. Introduction

> UMH Hub has a wide range of ports and connectors. 

Reviewer: The United Manufacturing Hub has a wide variety of connectors and therefore offers maximum connectivity to different machines and systems.

> {{< imgproc logos_data_sources Fit "800x800" >}}{{< /imgproc >}}

Reviewer: This cross out. And the subline "could contain sensitive information".
It is best to use "www.draw.io" for the graphic.
Insert the Weihenstephaner standard logo and the logos from our website in a "chaotic order".

![WS](https://user-images.githubusercontent.com/77407527/144581210-7f3b051e-1f43-4633-bc9a-01f4e03f52b6.png)
<img width="249" alt="image" src="https://user-images.githubusercontent.com/77407527/144581303-c22d98a5-12cc-46ee-9574-a5bdffaf1889.png">

> These are:

Reviewer: Here are a few selected examples

- OPC/UA ([documentation for this node](https://flows.nodered.org/node/node-red-contrib-opcua))
- Siemens S7 ([documentation for this node](https://flows.nodered.org/node/node-red-contrib-s7))
- TCP/IP ([documentation for this node](https://flows.nodered.org/flow/bed6f676d088670d7e1bc298943338b5))
- Rest API  ([documentation for this node](https://cookbook.nodered.org/http/create-an-http-endpoint))
- Modbus  ([documentation for this node](https://flows.nodered.org/node/node-red-contrib-modbus))
- MQTT ([documentation for this node](https://cookbook.nodered.org/mqtt/))

Via Node-RED, these data sources can be read out easily or, in the case of IO-Link, are even read out automatically and made available via MQTT. 

Using these data points and our data model, the data can then be converted into a standardized data model. This is done independently of machine type and manufacturer and can be made usable for various data services, e.g. Grafana.

> ## 2. Node-RED

Reviewer: ```suggestion
## 2. Extracting and preparing data with the help of Node-RED
```

To extract and preprocess the data from different data sources, we use the open source software Node-RED. Node-RED is a low-code programming for event-driven applications. 

Originally, Node-RED comes from the area of smart home and programming implementations, but is also used in manufacturing more frequently. For more information feel free to check [this article](https://docs.umh.app/docs/concepts/node-red-in-industrial-iot/).

> In the following, the procedure for creating a Node-RED flow is described in detail using **two examples**. The first example is about a cutting machine which will be retrofitted by external sensor technology. The second example deals with the connection of already existing sensors from with our system in order to visualize the acquired data.

Reviewer: ```suggestion
In the following, the procedure for creating a Node-RED flow is described in detail using **two examples**.

1. Here we will retrofit a cutting machine with the help of external sensors.
2. In the second example, we will use OPC/UA to read a warping machine data directly from the PLC.
```

### 1st example: Connecting external sensor technology

You would like to determine the output and machine condition of a cutting machine.

Used Sensors:

> - Light Barrier
> - Button Bar

Reviewer: ```suggestion
- Button Bar
```
```suggestion
- Light Barrier
- Button Bar
- Capacitive sensor
```

With the help of Sensorconnect, sensors can be connected quickly and easily via an IFM gateway. The sensor values are automatically extracted from the software stack and made available via [MQTT](https://docs.umh.app/docs/concepts/mqtt/). 

> *1. Connect Sensors*

Reviewer: ```suggestion
#### 1. Connect Sensors
```

TODO: Picture or illustration of how to connect sensors

> *2. Make the connected sensors visible to our system*

Reviewer: ```suggestion
#### 2. Make the connected sensors visible to our system
```

Based on an IP address, which is assigned to each sensor (or gateway?), the sensor can be integrated into our system. Only then is it possible to read out sensor values. The adaptation of the IP range is required.

TODO: ADAPTION OF IP-RANGE

> *3. Creating a flow*

Reviewer: ```suggestion
#### 3. Creating a Node-RED flow*
```

> In Node-RED basically three pieces of information must be communicated to the system.

Reviewer: In order to contextualise the resulting data points with the help of the United Manufacturing Hub, 3 identifiers are necessary:

> - The customer ID to be assigned to the asset: *customerID*

Reviewer: ```suggestion
- The customer ID to be assigned to the asset: *customerID* (Default value for the development setup "factoryinsight")
```

> - The location where the asset is located: *location*

Reviewer: ```suggestion
- The location where the asset is located: *location* (Can be chosen freely)
```

> - The name of the asset: *AssetID*

Reviewer: ```suggestion
- The name of the asset: *AssetID* (Can be chosen freely)
```
